File: scripts/lib/load_env.py
"""Load repo-root .env into os.environ (zero deps)."""
from __future__ import annotations

import os
from pathlib import Path

ROOT = Path(__file__).resolve().parents[2]


def load_env(*extra_files: str | Path) -> None:
    paths = [ROOT / ".env", ROOT / "backend" / ".env", *[Path(p) for p in extra_files]]
    for path in paths:
        if not path.is_file():
            continue
        for raw in path.read_text(encoding="utf-8").splitlines():
            line = raw.strip()
            if not line or line.startswith("#") or "=" not in line:
                continue
            key, _, val = line.partition("=")
            key = key.strip()
            val = val.strip()
            if (val.startswith('"') and val.endswith('"')) or (
                val.startswith("'") and val.endswith("'")
            ):
                val = val[1:-1]
            os.environ.setdefault(key, val)


def require_env(name: str) -> str:
    val = os.environ.get(name)
    if not val:
        raise SystemExit(
            f"Missing required env var: {name}. Add it to the repo root .env (see .env.example)."
        )
    return val


def env_or(name: str, fallback: str) -> str:
    return os.environ.get(name) or fallback

File: scripts/lib/load-env.mjs
/**
 * Load root (and optional backend) .env into process.env.
 * Zero deps — simple KEY=VALUE parser.
 */
import { readFileSync, existsSync } from "fs";
import { dirname, join } from "path";
import { fileURLToPath } from "url";

const ROOT = join(dirname(fileURLToPath(import.meta.url)), "..", "..");

function parseEnvFile(filePath) {
  if (!existsSync(filePath)) return;
  const text = readFileSync(filePath, "utf8");
  for (const raw of text.split(/\r?\n/)) {
    const line = raw.trim();
    if (!line || line.startsWith("#")) continue;
    const eq = line.indexOf("=");
    if (eq <= 0) continue;
    const key = line.slice(0, eq).trim();
    let val = line.slice(eq + 1).trim();
    if (
      (val.startsWith('"') && val.endsWith('"')) ||
      (val.startsWith("'") && val.endsWith("'"))
    ) {
      val = val.slice(1, -1);
    }
    if (process.env[key] === undefined) process.env[key] = val;
  }
}

export function loadEnv(extraFiles = []) {
  parseEnvFile(join(ROOT, ".env"));
  parseEnvFile(join(ROOT, "backend", ".env"));
  for (const f of extraFiles) parseEnvFile(f);
  return process.env;
}

export function requireEnv(name) {
  const v = process.env[name];
  if (v === undefined || v === "") {
    throw new Error(
      `Missing required env var: ${name}. Add it to the repo root .env (see .env.example).`
    );
  }
  return v;
}

export function envOr(name, fallback) {
  const v = process.env[name];
  return v === undefined || v === "" ? fallback : v;
}

export { ROOT };
